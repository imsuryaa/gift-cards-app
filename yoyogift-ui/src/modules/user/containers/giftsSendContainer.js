import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchSentCards } from './../state/actions/index';
import GiftsSent from '../components/giftsSend.js';
import CircularProgress from '@material-ui/core/CircularProgress';
import { Redirect } from 'react-router-dom';

export class GiftsSendContainer extends Component {

  componentDidMount() {
    if(this.props.user)
    this.props.fetchSentCards(this.props.user.email);
  }
  componentDidCatch(error, info) {
    console.log(error)
  }
  render() {
    if (this.props.isLoggedIn) {
      if(!(this.props.sentCards) || (this.props.sentCards.length < 0)) {
        return <CircularProgress style={{marginLeft: '50%', marginTop: '10%'}} />
      } else if (this.props.sentCards.length === 0) {
        return <h2 style={{
          height: '40px',
          background: '#b3d9f7',
          color: 'white',
          textAlign: 'center',
          verticalAlign: 'middle',
          paddingTop: '5px',
          fontSize: '20px',
          fontWeight: '500'
        }}>
        NO DATA
        </h2>
        } else {
        return (
          <div>
            <GiftsSent data={this.props.sentCards} />
          </div>
        )
      }    
    } else {
        return <Redirect to="/" />
    }
  }
}

const mapStateToProps = (state) => {
    return {
        user:state.login.detailsObject,
        isLoggedIn: state.login.loginStatus,
        sentCards: state.users.cards
    }
}

export default connect(mapStateToProps, {fetchSentCards})(GiftsSendContainer);
