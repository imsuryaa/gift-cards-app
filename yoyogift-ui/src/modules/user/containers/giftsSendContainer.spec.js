import React from "react";
import { mount } from "enzyme";

import { GiftsSendContainer } from "./giftsSendContainer";

describe("Gifts Sent Test Suite", () => {
  it('should show circular progress', () => {
    const wrapper = mount(<GiftsSendContainer 
    isLoggedIn = {true}
    sentCards={undefined}
    fetchSentCards={jest.fn()}
    />);

    expect(wrapper.find('div').length).toBe(1);
  });

  it('should show header with information', () => {
    const wrapper = mount(<GiftsSendContainer 
    isLoggedIn = {true}
    sentCards={[]}
    fetchSentCards={jest.fn()}
    />);

    expect(wrapper.find('h2').length).toBe(1);
    expect(wrapper.find('h2').text()).toBe("NO DATA");
  });

  it('should show table with header and data', () => {
    //Mocks
    const data = [
        {
          cardName: 'A',
          cardPoints: 1,
          receiverEmail: 'jalagamsurya.loriini@gmail.com',
          cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
          cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        },
        {
          cardName: 'B',
          cardPoints: 2,
          receiverEmail: 'jalagamsurya.loriini@gmail.com',
          cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
          cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        },
        {
          cardName: 'C',
          cardPoints: 3,
          receiverEmail: 'jalagamsurya.loriini@gmail.com',
          cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
          cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        }
    ];
    const wrapper = mount(<GiftsSendContainer 
    isLoggedIn = {true}
    sentCards={data}
    fetchSentCards={jest.fn()}
    />);
    
    expect(wrapper.find('tr').length).toBe(4);
    expect(wrapper.find('th').length).toBe(5);
    expect(wrapper.find('td').length).toBe(15);
  });
  
  it('should render the expected HTML', () => {
    expect(mount(<GiftsSendContainer 
    isLoggedIn = {true}
    sentCards={undefined}
    fetchSentCards={jest.fn()}
    />
    ).html()).toMatchSnapshot();
  });

  it('should render the expected HTML', () => {
    expect(mount(<GiftsSendContainer 
    isLoggedIn = {true}
    sentCards={[]}
    fetchSentCards={jest.fn()}
    />
    ).html()).toMatchSnapshot();
  });

  it('should render the expected HTML', () => {
    //Mocks
    const data = [
        {
          cardName: 'A',
          cardPoints: 1,
          receiverEmail: 'jalagamsurya.loriini@gmail.com',
          cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
          cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        },
        {
          cardName: 'B',
          cardPoints: 2,
          receiverEmail: 'jalagamsurya.loriini@gmail.com',
          cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
          cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        },
        {
          cardName: 'C',
          cardPoints: 3,
          receiverEmail: 'jalagamsurya.loriini@gmail.com',
          cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
          cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)"
        }
    ];

    expect(mount(<GiftsSendContainer 
    isLoggedIn = {true}
    sentCards={data}
    fetchSentCards={jest.fn()}
    />
    ).html()).toMatchSnapshot();
  });

});
