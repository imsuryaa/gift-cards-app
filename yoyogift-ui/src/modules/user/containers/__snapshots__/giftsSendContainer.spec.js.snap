// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts Sent Test Suite should render the expected HTML 1`] = `"<div class=\\"MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3\\" style=\\"width: 40px; height: 40px; margin-left: 50%; margin-top: 10%;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-6\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div>"`;

exports[`Gifts Sent Test Suite should render the expected HTML 2`] = `"<h2 style=\\"height: 40px; background: rgb(179, 217, 247); color: white; text-align: center; vertical-align: middle; padding-top: 5px; font-size: 20px; font-weight: 500;\\">NO DATA</h2>"`;

exports[`Gifts Sent Test Suite should render the expected HTML 3`] = `"<div><div class=\\"MuiPaper-root-13 MuiPaper-elevation2-17 MuiPaper-rounded-14 GiftsSend-root-11\\"><table class=\\"MuiTable-root-40 GiftsSend-table-12\\"><thead class=\\"MuiTableHead-root-41\\"><tr class=\\"MuiTableRow-root-42 MuiTableRow-head-45\\"><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">CARD NAME</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">POINTS</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">SENT TO</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">ISSUE DATE</th><th class=\\"MuiTableCell-root-47 MuiTableCell-head-48\\" scope=\\"col\\">EXPIRY DATE</th></tr></thead><tbody class=\\"MuiTableBody-root-59\\"><tr class=\\"MuiTableRow-root-42\\"><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">A</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">1</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">jalagamsurya.loriini@gmail.com</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td></tr><tr class=\\"MuiTableRow-root-42\\"><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">B</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">jalagamsurya.loriini@gmail.com</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td></tr><tr class=\\"MuiTableRow-root-42\\"><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">C</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">3</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">jalagamsurya.loriini@gmail.com</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td><td class=\\"MuiTableCell-root-47 MuiTableCell-body-49\\">2019-05-19</td></tr></tbody></table></div></div>"`;
